### PR TITLE
Add handling of non-JSON pacts

### PR DESCRIPTION
**Pull request by elliotwms-f3**
Co-authored-by: jose martins <jose.martins@form3.tech>

Commits will be squashed

This PR adds capability to handle also `text/plain` media type requests. Changes:
- when registering pact interaction or handling request via pact-proxy:
  - when the `Content-Type` header is not defined the default media type used is `text/plain`. Same as pact does.
  - when the `Content-Type` header is set to empty/invalid value/non JSON or text value the request fails with error
- for `application/json` media type the matching is the same as before
- for `text/plain` media type the matching is performed on the whole request body

So from now on all the requests/interactions used with pact-proxy need to have `Content-Type` header set to `application/json` or `text/plain` or empty/missing (defaults to `text/plain`). Otherwise the request will fail and pact-proxy will not process it. We had to force this in order to provide proper matching etc.

Proof of work
tests & hooked up to swift-gateway tests that now contain also plaintext requests and all the tests in swift-gw were happy :confetti_ball:

### CODE REVIEW

File: internal/app/pactproxy/interaction.go [hunk 1]
```diff
@@ -104,6 +110,30 @@ func LoadInteraction(data []byte, alias string) (*interaction, error) {
 	return interaction, nil
 }
 
+func isJSONRequest(request map[string]interface{}) (bool, error) {
+	headers, hasHeaders := request["headers"]
+	if !hasHeaders {
+		return false, nil
+	}
+
+	parsed, ok := headers.(map[string]string)
```
Comment (josemartins-form3) [inline internal/app/pactproxy/interaction.go:119]: http headers are normally considered `map[string][]string` don't know why but here's the [link](https://pkg.go.dev/net/http#Header)

File: internal/app/pactproxy/response_modification_writer.go [hunk 2]
```diff
@@ -7,36 +7,46 @@ import (
 )
 
 type ResponseModificationWriter struct {
-	res          http.ResponseWriter
-	interactions []*interaction
-	statusCode   int
+	res              http.ResponseWriter
+	interactions     []*interaction
+	originalResponse []byte
```
Comment (elliotwms-f3) [inline internal/app/pactproxy/response_modification_writer.go:12]: Is this change included in #17 separately? We should merge that first if so

Comment (plieskovsky-f3) [reply internal/app/pactproxy/response_modification_writer.go:12]: Yes exactly. We merged the branch `liesk-naveen-fix-response-modifications` to this one so we can test it in swift-gw. The plan is to first merge https://github.com/form3tech-oss/pact-proxy/pull/17 and then proceed with this PR (rebase etc.) :)

**Comment by plieskovsky-f3** [comment]
Closing in favour of https://github.com/form3tech-oss/pact-proxy/pull/18 - so we can make reviews and approve inside the team.

**Pull request closed without merging**